### PR TITLE
Replace `none()` with `discard()`

### PR DESCRIPTION
**Pull request by ryn5**
Renames none() to discard() to make room for a new list filtering none() step in https://github.com/apache/tinkerpop/pull/2385.

### CODE REVIEW

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/__.java [hunk 1]
```diff
@@ -806,6 +805,13 @@ public static <A> GraphTraversal<A, Set<?>> difference(final Object values) {
         return __.<A>start().difference(values);
     }
 
+    /**
+     * @see GraphTraversal#discard()
+     */
+    public static <A> GraphTraversal<A, A> discard() {
```
Comment (Cole-Greer) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/__.java:811]: I notice that there is no corresponding `none()` method being replaced here, is this needed?

**Review by Cole-Greer — COMMENTED** [review]
Thanks Ryan, these changes LGTM.

VOTE +1 (pending 1 minor comment below)

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/__.java (hunk 1 above)
Comment (ryn5) [reply gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/__.java:811]: This was originally added to pass a test that looks for steps being implemented in GraphTraversal, but after looking into it again it seems it should just be ignored as none() was previously, so I'll be adding "discard" as an ignore case and removing this function.

**Comment by vkagamlyk** [comment]
VOTE+1

**Comment by xiazcy** [comment]
LGTM. VOTE +1

**Pull request merged**